pkg/controller/machine/machine_controller.go

```diff
@@ -305,3 +341,35 @@ func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cl
 
 	return corev1Remote.Nodes().Delete(name, &metav1.DeleteOptions{})
 }
+
+// getMachines returns all of the Machine objects that belong to the same
+// cluster as the provided Machine
+func (r *ReconcileMachine) getMachines(ctx context.Context, machine *clusterv1.Machine) ([]*clusterv1.Machine, error) {
+	clusterName := machine.Labels[clusterv1.MachineClusterLabelName]
+	if clusterName == "" {
+		klog.Infof("Machine %q in namespace %q doesn't specify %q label, assuming nil cluster", machine.Name, machine.Namespace, clusterv1.MachineClusterLabelName)
+		return nil, nil
+	}
+
+	mlist := &clusterv1.MachineList{}
+	listOptions := &client.ListOptions{
+		Namespace: machine.Namespace,
+		LabelSelector: labels.Set(map[string]string{
+			clusterv1.MachineClusterLabelName: clusterName,
+		}).AsSelector(),
+	}
+
+	if err := r.Client.List(ctx, listOptions, mlist); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, errors.Wrapf(err, "Failed to list machines")
+	}
+
+	machines := make([]*clusterv1.Machine, len(mlist.Items))
+	for i := range mlist.Items {
+		machines[i] = &mlist.Items[i]
+	}
+
+	return machines, nil
```

Comment:
Can we return `mlist.Items`?

Reply:
`util.GetControlPlaneMachines` needs the pointers